src/marqo/core/inference/models/utils.py

```diff
@@ -0,0 +1,236 @@
+import os
+from io import BytesIO
+
+import certifi
+import numpy as np
+import pycurl
+import requests
+import torch
+import validators
+from PIL import Image, UnidentifiedImageError
+from requests.utils import requote_uri
+
+from marqo import marqo_docs
+from marqo.api.exceptions import InternalError
+from marqo.s2_inference.errors import ImageDownloadError
+from marqo.s2_inference.types import *
+from marqo.tensor_search.telemetry import RequestMetrics
+
+# TODO Merge this with the one in clip_utils in the future refactoring
+
+DEFAULT_HEADERS = {'User-Agent': 'Marqobot/1.0'}
+
+
+def get_allowed_image_types():
+    return {'.jpg', '.png', '.bmp', '.jpeg'}
+
+
+def _is_image(inputs: Union[str, List[Union[str, ImageType, ndarray]]]) -> bool:
```

Comment:
Can we use abstract model class? Utils are often bad design

Reply:
Rename to image_download.py as discussed